### PR TITLE
[Fix] Parameterized IDP_METADATA_URL and fixed return of getUserCompanies

### PR DESCRIPTION
**Pull request by michaeldisaro**
Parameterized IDP_METADATA_URL and fixed return of getUserCompanies

### CODE REVIEW

File: src/utils/attribute_authority.ts [hunk 1]
```diff
@@ -37,8 +37,8 @@ export const getUserCompanies = (
 
     TE.chain((arr) => {
       return arr.length > 0
-        ? TE.left(ResponseErrorForbiddenNotAuthorized)
-        : TE.of(arr);
+        ? TE.of(arr)
+        : TE.left(ResponseErrorForbiddenNotAuthorized);
     }),
```
Comment (BurnedMarshal) [inline src/utils/attribute_authority.ts:42]: What if we use a `fromPredicate` instead?
```suggestion
    TE.chainW(
      TE.fromPredicate(
        (_) => _.length > 0,
        () => ResponseErrorForbiddenNotAuthorized
      )
    ),
```

**Review by BurnedMarshal — COMMENTED** [review]
LGTM apart a minor suggestion

File: src/utils/attribute_authority.ts (hunk 1 above)
Comment (michaeldisaro) [reply src/utils/attribute_authority.ts:42]: Yes, I just fixed existing code, but you're right, it's a better solution to refactor since we are here.

Comment (michaeldisaro) [reply src/utils/attribute_authority.ts]: done in https://github.com/pagopa/hub-spid-login-ms/pull/54/commits/fd76da21c20a5323440aee87aa66c97e8e08323b 
I tried also for the ternary above this, but fromPredicate was not dividing the union of results...

File: yarn-error.log [hunk 2]
```diff
@@ -0,0 +1,6847 @@
+Arguments:
```
Comment (BurnedMarshal) [inline yarn-error.log:1]: Remove this file or add it to `.gitignore`

**Review by BurnedMarshal — CHANGES_REQUESTED** [review]

File: yarn-error.log (hunk 2 above)
Comment (michaeldisaro) [reply yarn-error.log]: Done in [3b1e9af](https://github.com/pagopa/hub-spid-login-ms/pull/54/commits/3b1e9af44a18828a6bc032b010cb0daddb71d562)

**Review by BurnedMarshal — APPROVED** [review]
LGTM

**Pull request merged**